src/test/java/com/getindata/connectors/http/internal/table/lookup/HttpLookupTableSourceITCaseTest.java

```diff
@@ -1058,14 +1192,269 @@ private void setUpServerBodyStub(
+            // No need to set up server stub for connection error test
+            // The test will use a non-existent port (9091)
+        } else {
+            // Setup for success test
+            if (StringUtils.isNullOrWhitespaceOnly(spec.methodName) || spec.methodName.equalsIgnoreCase("GET")) {
+                setupServerStub(wireMockServer);
+            } else {
+                setUpServerBodyStub(
+                        spec.methodName,
+                        wireMockServer,
+                        List.of(matchingJsonPath("$.id"), matchingJsonPath("$.id2"))
+                );
+            }
+        }
+    }
+
+    private String createLookupTableSql(TestSpec spec) {
+        StringBuilder sql = new StringBuilder();
+        sql.append("CREATE TABLE Customers (")
+                .append("id STRING,")
+                .append("id2 STRING,")
+                .append("msg STRING,")
+                .append("uuid STRING,")
+                .append("details ROW<")
+                .append("isActive BOOLEAN,")
+                .append("nestedDetails ROW<")
+                .append("balance STRING")
+                .append(">")
+                .append(">");
+
+        if (spec.useMetadata) {
+            sql.append(",")
+                    .append("errStr STRING METADATA FROM 'error_string',")
+                    .append("statusCode INTEGER METADATA FROM 'http_status_code',")
+                    .append("headers MAP<STRING, ARRAY<STRING>> METADATA from 'http_headers',")
+                    .append("completionState STRING METADATA from 'http_completion_state'");
+        }
+
+        sql.append(") WITH (")
+                .append("'format' = 'json',")
+                .append("'connector' = 'rest-lookup',");
+
+        if (!StringUtils.isNullOrWhitespaceOnly(spec.methodName)) {
+            sql.append("'lookup-method' = '").append(spec.methodName).append("',");
+        }
+
+        // URL with correct port for connection error test
+        if (spec.connectionError) {
+            sql.append("'url' = 'http://localhost:9091/client',");
+        } else {
+            sql.append("'url' = 'http://localhost:9090/client',");
+        }
+
+        sql.append("'gid.connector.http.source.lookup.header.Content-Type' = 'application/json',");
+
+        //if (spec.useMetadata) {
+        //    sql.append("'gid.connector.http.source.lookup.continue_on_error    '='true',");
+        //}
+        sql.append("'gid.connector.http.source.lookup.continue_on_error'='true',");
+        sql.append("'asyncPolling'='false')");
+
+        //sql.append("'asyncPolling' = '").append(spec.useAsync ? "true" : "false").append("',")
+        //    .append("'table.exec.async-lookup.buffer-capacity' = '50',")
+        //    .append("'table.exec.async-lookup.timeout' = '120s'")
+        //    .append(")");
+
+        return sql.toString();
+    }
+
+    private void assertResultsForSpec(TestSpec spec, Collection<Row> rows) {
```

Comment:
nit: I think this method should be defined above methods it uses.

Reply:
I just copied the way that the Kafka connector did it - so would be inclined to keep it the same. WDYT?